src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -259,20 +264,23 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 			$theme_json      = apply_filters( 'wp_theme_json_data_theme', new WP_Theme_JSON_Data( $theme_json_data, 'theme' ) );
 			$theme_json_data = $theme_json->get_data();
 			static::$theme   = new WP_Theme_JSON( $theme_json_data );
-		}
```

Comment:
This looks like an oversight. I can't think of a reason this would have been desired (check for the parent even if the child had already been computed and blocks didn't change).